Docs: Configure security hardening

KristianGrafana opened the pull request:
What is this feature?
This PR adds documentation for how to configure additional security hardening to the Grafana instance.

Why do we need this feature?
So Grafana admins know how to further secure the instance to protect against certain attacks (XSS and CSRF mainly). 

Who is this feature for?
Grafana admins and/or security engineers.

KristianGrafana committed:
Docs: Configure security hardening

KristianGrafana committed:
Docs: Configure security hardening

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/grafana/grafana?pullRequest=58738) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/grafana/grafana?pullRequest=58738) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/grafana/grafana?pullRequest=58738) it.</sub>

kalleep committed:
Auth: Always include oauth and saml settings for frontend (#58705)

* Auth: Always include oauth and saml settings

KristianGrafana committed:
Linting with Prettier

KristianGrafana committed:
Linting with Prettier

papagian committed:
Chore: Restore folder properties (#58743)

* Chore: Fix folder URL

* Restore more folder properties

* Fixup

torkelo committed:
Scene: Variables and support for declaring variable dependencies and getting notified or re-rendered when they change (#58299)

* Component that can cache and extract variable dependencies

* Component that can cache and extract variable dependencies

* Updates

* Refactoring

* Lots of refactoring and iterations of supporting both re-rendering and query re-execution

* Updated SceneCanvasText

* Updated name of file

* Updated

* Refactoring a bit

* Added back getName

* Added comment

* minor fix

* Minor fix

* Merge fixes

* Merge fixes

* Some review fixes

* Updated comment

* Added forceRender function

* Add back fail on console log

ashharrison90 committed:
Navigation: Add quick actions button (#58707)

* initial implementation for quick add

* add new isCreateAction prop on NavModel

* adjust separator margin

* switch to primary button

* undo changes to plugin.json

* remove unused props from interface

* use a consistent dropdown overlay type

* memoize findCreateActions

* add prop description

* use a function so that menus are only rendered when the dropdown is open

JoaoSilvaGrafana committed:
Search: Fixes issue with Recent/Starred section always displaying "General" folder (#58746)

leandro-deveikis committed:
Chore: Update version (#58750)

mgyongyosi commented on docs/sources/setup-grafana/configure-security/configure-security-hardening.md:
I think the titles should be swapped: L76 with L88.

KristianGrafana replied:
Thanks, good catch, fixed in https://github.com/grafana/grafana/pull/58738/commits/c401ffc47c9961af4291d05807759b6aef047786

chri2547 commented on docs/sources/setup-grafana/configure-security/configure-security-hardening.md:
which toml file?

KristianGrafana replied:
I added a note in the beginning regarding which file to be modified.

chri2547 commented on docs/sources/setup-grafana/configure-security/configure-security-hardening.md:
```suggestion
# Set to true if you host Grafana behind HTTPS. The default value is false.
```

ashharrison90 committed:
Revert "Chore: move to node 18 (#58570)" (#58754)

This reverts commit 0a9129cf90a7134411f8aeca1c0bd56b112e9371.

Commit:
plugindef: Move pluginmeta out of coremodels as standalone thema lineage (#56765)

* Get pluginmeta mostly moved over to pkg/plugins/plugindef

* Remove dead func

* Fix up pfs, use sync.Once in plugindef

* Update to latest thema

* Chase Endec->Codec conversion in Thema

* Comments on slash header gen; use ToSlash

* Also generate JSON schema for plugindef

* Generate JSON Schema as well

* Fix slot loading from kindsys cue decls

* Remove unused vars

* skip generating plugin.schema.json for now

Co-authored-by: Marcus Efraimsson <marcus.efraimsson@gmail.com>

gelicia committed:
Refactor SplitPaneWrapper to be more centralized component, refactor PanelEditor to use it (#58380)

* Move layout to paneleditor, make SplitPaneWrapper more generic

* Read/write the size ratio in local storage

* Add min height to enable scrollbar

* Enable show/hide panel options

* Change back variable name

KristianGrafana committed:
Changed sub titles, added more descriptive text and fixed issues from git comments

KristianGrafana committed:
Changed sub titles, added more descriptive text and fixed issues from git comments

eledobleefe committed:
Internationalization: Translate VariableInput and VariableOptions components (#58748)

grafanabot committed:
Changelog: Updated changelog for 9.3.0-beta1 (#58785)

dsotirakis committed:
latest.json: Update `latest.json` to 9.3.0-beta1 (#58788)

Update latest.json

KristianGrafana committed:
Add link to configuration file location.

KristianGrafana committed:
Add link to configuration file location.

mckn committed:
Docs: How to add plugin interaction tracking (#58652)

* docs for plugin interaction tracking.

* Update docs/sources/developers/plugins/add-anonymous-usage-reporting.md

Co-authored-by: Marcus Efraimsson <marcus.efraimsson@gmail.com>

* Adding query type

* Fixed spelling issue

Co-authored-by: Marcus Efraimsson <marcus.efraimsson@gmail.com>

tolzhabayev committed:
Fix: Bump-version action regex pattern to work with beta1 (#58805)

Fixing bump version regex

Elfo404 committed:
Explore: Refactor ExploreGraph (#58660)

* WIP

* revert collapse changes

* use HorizontalGroup instead of custom styles

* fix tests

* use import aliases

torkelo committed:
Scene: Variables interpolation formats and multi value handling  (#58591)

* Component that can cache and extract variable dependencies

* Component that can cache and extract variable dependencies

* Updates

* Refactoring

* Lots of refactoring and iterations of supporting both re-rendering and query re-execution

* Updated SceneCanvasText

* Updated name of file

* Updated

* Refactoring a bit

* Added back getName

* Added comment

* minor fix

* Minor fix

* Merge fixes

* Scene variable interpolation progress

* Merge fixes

* Added all format registeries

* Progress on multi value support

* Progress on multi value support

* Updates

* Progress on scoped vars

* Fixed circular dependency

* Updates

* Some review fixes

* Updated comment

* Added forceRender function

* Add back fail on console log

* Update public/app/features/scenes/variables/interpolation/sceneInterpolator.test.ts

* Moving functions from SceneObjectBase

* fixing tests

* Fixed e2e

Co-authored-by: Dominik Prokop <dominik.prokop@grafana.com>

leventebalogh committed:
AppRootPage: Render app plugins without pages (#58776)

fix: render app plugins that don't have a page in includes

eledobleefe committed:
Internationalization: Translate NavBar - 'Search dashboard' menu item (#58815)

eledobleefe committed:
Internationalization: Translate ShareSnapshot label (#58802)

eledobleefe committed:
Internationalization: Translate 'Hide / show legend' of PanelHeaderMenuItem (#58800)

zuchka committed:
Admin: Fix broken links to image assets in email templates (#58729)

fix broken links to image assets

chri2547 commented on docs/sources/setup-grafana/configure-security/configure-security-hardening.md:
```suggestion
> **Note:** These settings are available in the [grafana.ini configuration file]({{< relref "../configure-grafana/#configuration-file-location" >}}). To apply changes to the configuration file, restart the Grafana server.
```

KristianGrafana committed:
Update docs/sources/setup-grafana/configure-security/configure-security-hardening.md

Co-authored-by: Christopher Moyer <35463610+chri2547@users.noreply.github.com>

KristianGrafana committed:
Update docs/sources/setup-grafana/configure-security/configure-security-hardening.md

Co-authored-by: Christopher Moyer <35463610+chri2547@users.noreply.github.com>

torkelo committed:
Variables: Use new format registry from templateSrv (#58813)

* Variables: Use new format registry from templateSrv

* Updated comment

* Fixed e2e

dsotirakis committed:
CI: Cleanup `e2e` tests dependencies (#58829)

Cleanup end to end tests dependencies

yuri-tceretian committed:
kindsys: Make kind generators run in Windows (#58794)

torkelo committed:
Scenes: Share factory function for test data query runner (#58816)

gamab committed:
RBAC: Cover plugin includes (#57582)

* RBAC: Add action to plugin includes

* Adding the feature toggle check

* Cue update

* Extract include access control to method

* Suggestion to prevent log when RBAC is disabled

Co-authored-by: ievaVasiljeva <ieva.vasiljeva@grafana.com>

* Rename IsRBACReady to RequireRBACAction

Co-authored-by: ievaVasiljeva <ieva.vasiljeva@grafana.com>

kalleep committed:
Teams: Support paginating and filtering more then 1000 teams (#58761)

* TeamList: break out rows to its own component

* TeamsState: Add total count

* TeamList: Remove teamsCount prop

* TeamList: Restructure code and use count from backend response

* TeamList: calculate total pages using totalCount

* TeamList: Rename to state to currentPage and the reducer to
setCurrentPage

* TeamList: remove wrapper functions

* TeamList: rewrite as a functional component

* TeamList: export components for test

* TeamList: pass limit, page and query to backend

* TeamList: Rename properties in state and create actions for page and
query change

* TeamList: Add flag to control if EmptyList banner should render

L-M-K-B committed:
Explore: A11y of range slider in query history (#58708)

refactor: remove temporary setting

ryantxu committed:
QueryData: fix header parsing to support expressions (#58826)

fixes #58821

ArturWierzbicki committed:
Chore: skip flaky tests (#58835)

skip flaky tests

kylebrandt committed:
CodeOwners: (Chore) Add Server Side Expressions (SSE) (#58841)

CodeOwners: Add Server Side Expressions (SSE)
owned by observability-metrics

kylebrandt committed:
SSE: Keep value name from numeric table (#58831)

fixes #48868

ivanortegaalba committed:
Build: Disable flaky RuleEditor frontend test (#58844)

gamab committed:
RBAC: Add tests on AddAppLinks (#58843)

* RBAC: Add tests on AddAppLinks

zuchka committed:
DataLinks: Fix double dollar-sign bug in data-links editor (#58096)

* fix template variable bug

* fix bug when adding vars before existing vars

* take 2--remove includeDollarSign logic

* take 3-add includeDollarSign logic for template vars

jcalisto committed:
Middleware: Add CSP Report Only support (#58074)

* Middleware: Add CSP Report Only support

* Update docs/sources/setup-grafana/configure-grafana/_index.md

Co-authored-by: Christopher Moyer <35463610+chri2547@users.noreply.github.com>

* Update docs/sources/setup-grafana/configure-grafana/_index.md

Co-authored-by: Christopher Moyer <35463610+chri2547@users.noreply.github.com>

* Update csp documentation wording

* Update conf/sample.ini

Co-authored-by: Dave Henderson <dave.henderson@grafana.com>

* Update docs/sources/setup-grafana/configure-grafana/_index.md

Co-authored-by: Dave Henderson <dave.henderson@grafana.com>

* Update docs/sources/setup-grafana/configure-grafana/_index.md

Co-authored-by: Dave Henderson <dave.henderson@grafana.com>

* Update docs/sources/setup-grafana/configure-grafana/_index.md

Co-authored-by: Dave Henderson <dave.henderson@grafana.com>

* Update pkg/middleware/csp.go

Co-authored-by: Dave Henderson <dave.henderson@grafana.com>

Co-authored-by: Christopher Moyer <35463610+chri2547@users.noreply.github.com>
Co-authored-by: Dave Henderson <dave.henderson@grafana.com>

zuchka committed:
Query Editor: Hide overflow for long query names (#58840)

hides overflow for long query names

grafanabot committed:
Changelog: Updated changelog for 9.2.5 (#58856)

marefr committed:
Database: Adds support for enable/disable SQLite Write-Ahead Logging (WAL) via configuration (#58268)

Adds support for enable/disable SQLite Write-Ahead Logging (WAL) via configuration.
Enables SQLite WAL for E2E tests.

leandro-deveikis committed:
Chore: update latest.json to 9.2.5 (#58860)

Commit:
codejen: Update to latest codejen (#58866)

papagian committed:
Quota: Fix failure when checking session limits (#58865)

lpskdl committed:
Folder: Replace folderId with folderUid (#58393)

* support folderuid in FolderPicker

* support folderuid in unified alerting

* support folderuid when returning to view mode after editing a panel

* support folderuid when preselecting the folderpicker in dashboard general settings

* support folderuid when saving dashboard

* support folderuid when pre-selecting folderpicker in dashboard form

* support folderuid in routes when loading a dashboard

* support folderuid when saving dashboard json

* support folderuid when validating new dashboard name

* support folderuid when moving dashboard to another folder

* support folderuid on dashboard action buttons

* support folderuid when creating a new dashboard on an empty folder

* support folderuid when showing library panel modal

* support folderuid when saving library panel

* support folderuid when importing dashboard

* fixed broken tests

* use folderuid when importing dashboards

* remove commented line

* fix typo when comparing uid values

ashharrison90 committed:
Navigation: Change quick add styling (#58854)

Change quick add styling + allow for returning isOpen state to dropdown children

eleijonmarck committed:
[Docs] Edit terraform example and doc for file generation (#58822)

* docs: update terraform example and doc for file generation

* docs: updated the documnettaion to include the help command with fields included

* Update pkg/cmd/grafana-cli/commands/commands.go

* docs: add help command

dsotirakis committed:
grafana.com: Make `beta` and `test` releases not stable (#58883)

If version is beta or test, don't mark it as stable

dsotirakis committed:
CI: Replace `TAG` with `DRONE_TAG` in CI (#58894)

* Replace TAG with DRONE_TAG

* Fix variable call

* Replace remaining bits

* Bump grabpl version

KristianGrafana committed:
criterias => criteria

KristianGrafana committed:
criterias => criteria

Jguer committed:
Chore: Extract server lock error so it can be used with errors.As (#58899)

chore: extract server lock Error so it can be used with error.As

kalleep committed:
Oauth: Remove unused function for oauth implementations (#58887)

* Oauth: remove unused function

* Oauth: remove unused Oauth types

KristianGrafana committed:
Format with Prettier

ashharrison90 committed:
Navigation: rename Grafana Machine Learning to just Machine Learning (#58893)

rename Grafana Machine Learning to just Machine Learning

KristianGrafana committed:
Merge branch 'docs-security-hardening' of https://github.com/grafana/grafana into docs-security-hardening
